packages/vitest/src/node/plugins/index.ts

```diff
@@ -123,12 +123,18 @@ export async function VitestPlugin(options: UserConfig = {}, ctx = new Vitest('t
           delete viteConfigTest.alias
 
         // viteConfig.test is final now, merge it for real
+
+        function debug(obj: object) {
+          return Object.fromEntries(Object.entries(obj).filter(([k, _v]) => ['retry', 'bail', 'segfaultRetry'].includes(k)))
+        }
+        console.log('== configResolved (before)', [debug(options), debug(viteConfigTest)])
         options = deepMerge(
           {},
           configDefaults,
           viteConfigTest,
           options,
         )
+        console.log('== configResolved (after)', [debug(options)])
```

Comment:
When using cac's `default` value, the debug here looks like this:

```
== configResolved (before) [ { segfaultRetry: 0, bail: 0, retry: 0 }, { retry: 3, bail: 4 } ]
== configResolved (after) [ { retry: 0, bail: 0, segfaultRetry: 0 } ]
```

It shows `viteConfigTest` has no effect for `bail` and `retry` since `options` has precedence over `viteConfigTest`.